prov/tcp/src/xnet_av.c

```diff
@@ -97,26 +99,27 @@ static int xnet_mplex_av_insertsym(struct fid_av *av_fid, const char *node,
 	int ret;
 	struct fid_list_entry *item;
 	struct fid_av *subav_fid;
-	fi_addr_t sub_fi_addr;
+	fi_addr_t *sub_fi_addr;
 	struct xnet_mplex_av *av = container_of(av_fid, struct xnet_mplex_av,
 						util_av.av_fid.fid);
-
+	sub_fi_addr = calloc(nodecnt*svccnt, sizeof(fi_addr_t));
 	ofi_genlock_lock(&av->lock);
 	ret = ofi_ip_av_insertsym(&av->util_av.av_fid, node, nodecnt,
 				  service, svccnt, fi_addr, flags, context);
-	if (ret)
+	if (ret < nodecnt*svccnt)
 		goto out;
 	dlist_foreach_container(&av->subav_list, struct fid_list_entry,	item, entry) {
 		subav_fid = container_of(item->fid, struct fid_av, fid);
 		ret = fi_av_insertsym(subav_fid, node, nodecnt, service, svccnt,
-				      &sub_fi_addr, flags, context);
-		if (ret)
+				      sub_fi_addr, flags, context);
+		if (ret <= nodecnt*svccnt)
 			break;
```

Comment:
What is the expectation of the resulting AV when some of the addresses fail to insert?  If the function returns a value < nodecnt * svccnt due to L109 or L115, the state of the main AV and sub AVs are going to be inconsistent. Will the provider be able to handle the case that the application proceeds to use the AV for one of the addressed successfully inserted in the partial successful call?  This is not an issue specifically for this PR, but rather is related to the suv-AV approach.

Reply:
Yeah, this is a fairly hard problem to handle I think.  If the av and sub-av are out of sync, it's basically a fatal error.  I suppose if the main av or sub-av fail to fully insert, we may need to roll back and fail the whole thing (for fi_av_insert, too).